### PR TITLE
The unit test development for the Core project is continuing.

### PR DESCRIPTION
**Pull request by sinanhaci**

### CODE REVIEW

**Comment by sinanhaci** [comment]
.

**Comment by sinanhaci** [comment]
.

**Review by KorayLiman — APPROVED** [review]

File: assets/test_image_1.jpeg
Comment (ozverimobile) [inline assets/test_image_1.jpeg:1]: Image size is too large. We may use another image with kilobytes instead.

File: lib/src/common/extensions/string_extensions/string_extensions.dart [hunk 1]
```diff
@@ -12,4 +12,10 @@ extension StringExtensionNullable on String? {
 
   /// Returns true if string is not null or not empty
   bool get isNotNullAndNotEmpty => this != null && this!.isNotEmpty;
+
+  /// returns null if string is empty
+  String? toNullIfEmpty() => (this ?? '') == '' ? null : this;
```
Comment (ozverimobile) [inline lib/src/common/extensions/string_extensions/string_extensions.dart:17]:
```suggestion
  String? get toNullIfEmpty => (this ?? '') == '' ? null : this;
```

File: lib/src/utils/input_formatter/input_formatters.dart [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import 'package:flutter_core/flutter_core.dart';
+
+class CoreInputformatter {
```
Comment (ozverimobile) [inline lib/src/utils/input_formatter/input_formatters.dart:3]:
```suggestion
abstract class CoreInputformatter {
```

File: lib/src/utils/input_formatter/input_formatters.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  CoreInputformatter._();
```
Comment (ozverimobile) [inline lib/src/utils/input_formatter/input_formatters.dart:4]:
```suggestion

```

File: lib/src/widgets/text_field/credit_card_security_code_text_field_impl.dart [hunk 4]
```diff
@@ -53,11 +57,12 @@ class CoreCreditCardSecurityCodeTextField extends StatelessWidget {
...
       keyboardType: TextInputType.number,
       autofillHints: autofillHints,
+      buildCounter: buildCounter,
       inputFormatters: [
-        LengthLimitingTextInputFormatter(4),
+        FilteringTextInputFormatter.digitsOnly,
       ],
-      buildCounter: buildCounter,
       decoration: InputDecoration(
+        counter: const CoreSizedBox.shrink(),
```
Comment (ozverimobile) [inline lib/src/widgets/text_field/credit_card_security_code_text_field_impl.dart:65]:
```suggestion
        counter: emptyBox,
```

**Review by ozverimobile — CHANGES_REQUESTED** [review]

**Review by huseyinozkan — APPROVED** [review]

**Review by ozverimobile — APPROVED** [review]

**Pull request merged**